src/backend/src/services/work-packages.services.ts

```diff
@@ -576,4 +578,29 @@ export default class WorkPackagesService {
       }
     });
   }
+
+  static async getBlockingWorkPackages(wbsNum: WbsNumber): Promise<WorkPackage[]> {
+    const { carNumber, projectNumber, workPackageNumber } = wbsNum;
+
+    if (workPackageNumber === 0) throw new HttpException(400, `${wbsPipe(wbsNum)} is not a valid work package WBS!`);
+
+    const workPackage = await prisma.work_Package.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      ...workPackageQueryArgs
+    });
+
+    if (!workPackage) throw new NotFoundException('Work Package', wbsPipe(wbsNum));
+
+    const blockingWbsElementIds = await getBlockingWbsElementIds(workPackage);
+
+    const blockingWps = await getWorkPackages(blockingWbsElementIds);
+
+    return blockingWps.map((wp) => workPackageTransformer(wp!));
```

Comment:
blockingWorkPackages.map(workPackageTransformer), also the fact that you need the `!` is very lazy coding